app/models/shipit/pull_request.rb

```diff
@@ -0,0 +1,184 @@
+module Shipit
+  class PullRequest < ApplicationRecord
+    WAITING_STATUSES = %w(fetching pending).freeze
+    InvalidTransition = Class.new(StandardError)
+
+    class StatusChecker < Status::Group
+      def initialize(commit, statuses, deploy_spec)
+        @deploy_spec = deploy_spec
+        super(commit, statuses)
+      end
+
+      private
+
+      attr_reader :deploy_spec
+
+      def reject_hidden(statuses)
+        statuses.reject { |s| ignored_statuses.include?(s.context) }
+      end
+
+      def reject_allowed_to_fail(statuses)
+        statuses.reject { |s| ignored_statuses.include?(s.context) }
+      end
+
+      def ignored_statuses
+        deploy_spec.try!(:pull_request_ignored_statuses) || []
+      end
+
+      def required_statuses
+        deploy_spec.try!(:pull_request_required_statuses) || []
+      end
+    end
+
+    belongs_to :stack
+    belongs_to :head, class_name: 'Shipit::Commit'
+    belongs_to :merge_requested_by, class_name: 'Shipit::User'
+
+    validates :number, presence: true, uniqueness: {scope: :stack_id}
+
+    scope :waiting, -> { where(merge_status: WAITING_STATUSES) }
+    scope :pending, -> { where(merge_status: 'pending') }
+    scope :to_be_merged, -> { pending.order(merge_requested_at: :asc) }
+
+    state_machine :merge_status, initial: :fetching do
```

Comment:
Ultimately it will be documented in the `README.md`, but for now it's an experimental feature marked as alpha that is likely to change in the future. It won't be documented until it's dimmed stable enough.